File: src/ui/src/models/api/Action/GetActionsItemModel.ts
import { BackendActionModelConfiguration } from './BackendActionModelConfiguration';
import { ToolProviderApplicationType } from '../provider/ToolProviderApplicationType';

/**
 * Model for an action item in the list view
 */
export interface ActionItem {
  /**
   * The unique identifier for the action
   */
  id: string;
  
  /**
   * The name of the action
   */
  name: string;
  
  /**
   * The description of the action
   */
  description: string;
  
  /**
   * The icon for the action
   */
  icon?: string;
  
  /**
   * The system prompt IDs associated with this action
   */
  systemPromptIds: string[];
  
  /**
   * The user prompt IDs associated with this action
   */
  userPromptIds: string[];
  
  /**
   * The allowed tools for this action
   */
  allowedTools: ToolProviderApplicationType[];
  
  toolProviderApplicationTypes?: ToolProviderApplicationType[];
  
  /**
   * The action model configuration
   */
  actionModelConfiguration: BackendActionModelConfiguration;
  
  /**
   * The timestamp when the action was created
   */
  createdAt: Date | string;
  
  /**
   * The timestamp when the action was last updated
   */
  updatedAt: Date | string;
}

File: src/backend/DonkeyWork.Chat.Api/Models/Action/GetActionsItemModel.cs
// ------------------------------------------------------
// <copyright file="GetActionsItemModel.cs" company="DonkeyWork.Dev">
// Provided as is by (c) DonkeyWork.Dev. No warranty or support is given.
// </copyright>
// ------------------------------------------------------

using DonkeyWork.Chat.Common.Models.Actions;
using DonkeyWork.Chat.Common.Models.Providers.Tools;

namespace DonkeyWork.Chat.Api.Models.Action;

/// <summary>
/// Model for an action item in the list.
/// </summary>
public class GetActionsItemModel
{
    /// <summary>
    /// Gets or sets the id.
    /// </summary>
    public Guid Id { get; set; }

    /// <summary>
    /// Gets or sets the name.
    /// </summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the description.
    /// </summary>
    public string Description { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the icon.
    /// </summary>
    public string? Icon { get; set; }

    /// <summary>
    /// Gets or sets the system prompt ids.
    /// </summary>
    public List<Guid> SystemPromptIds { get; set; } = [];

    /// <summary>
    /// Gets or sets the user prompt ids.
    /// </summary>
    public List<Guid> UserPromptIds { get; set; } = [];

    /// <summary>
    /// Gets or sets the allowed tools.
    /// </summary>
    public List<ToolProviderApplicationType> AllowedTools { get; set; } = [];

    /// <summary>
    /// Gets or sets the action model configuration.
    /// </summary>
    public ModelConfiguration ModelConfiguration { get; set; } = new ();

    /// <summary>
    /// Gets or sets the created at date.
    /// </summary>
    public DateTimeOffset CreatedAt { get; set; }

    /// <summary>
    /// Gets or sets the updated at date.
    /// </summary>
    public DateTimeOffset UpdatedAt { get; set; }
}
